app/views/products/_form.html.erb

```diff
@@ -0,0 +1,8 @@
+<%= form_for [ @vendor, @product ] do |f| %>
+  <%= f.label :name, "Name:" %>
+  <%= f.text_field :name, required: :true %>
+
+  <%= f.hidden_field :vendor_id, value: params[:vendor_id] %>
```

Comment:
Sorry I didn't check this earlier! Can you create products without a vendor? If you can't, then you'll probably want to have your url for the form be something like `/vendors/1234/products/new`. If it's set up like that, you should be able to grab the vendor from the params. Then you can find and instantiate the vendor from your database and do something like this:
```ruby
@vendor = Vendor.find(params[:vendor_id])
@vendor.products << Product.create(product_params)
```